### ISSUE TITLE
Caja does not show previews of text files on Ubuntu MATE 18.04 LTS

### ISSUE THREAD

**Issue by N0rbert** [mate-desktop/caja#1047]
#### Steps to reproduce the behaviour
1. Install Ubuntu MATE 18.04 LTS (or upgrade from 16.04 LTS)
2. Create text file somewhere (in home folder for example) with `echo text > ~/file.txt`.
3. Open Caja in home folder

#### Expected behaviour
* Caja shows preview (the contents) of `file.txt` in Icon view.

#### Actual behaviour
* Caja does not show preview of `file.txt`.

**Notes:**
1. all settings in Preview tab are set to *Local Only*
2. this happens on clean user profile and/or after `dconf reset -f /` and `gsettings reset-recursively  org.mate.caja`
3. Problem first seen on AskUbuntu ( https://askubuntu.com/q/1065825/66509 ).

#### MATE general version
1.20.1 as in Ubuntu MATE 18.04 LTS.

#### Package version
1.20.2-4ubuntu1

#### Linux Distribution
Ubuntu MATE 18.04 LTS

#### Link to downstream report of your Distribution
https://bugs.launchpad.net/ubuntu/+source/caja/+bug/1787368

**Comment by lukefromdc** [comment]
This is not working here with Caja 1.21 from git master as of 8-16-2018 on Debian Unstable either

**Comment by monsta** [comment]
I recall this stopped working after migration to GTK+3.

Some Nemo bug reports give some hints:
https://github.com/linuxmint/nemo/issues/949
https://github.com/linuxmint/nemo/issues/954
https://github.com/linuxmint/nemo/issues/1089
https://github.com/linuxmint/nemo/issues/1282

**Comment by vkareh** [comment]
GTK+ dropped this feature entirely, we would need to create a small library that implements this feature and maintain it independently of GTK+

Maybe we should start such a library to keep a handful of old GTK+ features that non-GNOME desktops rely on. We could get people from MATE, Cinnamon, Elementary, XFCE, etc involved.

**Comment by lukefromdc** [comment]
That library could be named gtk-legacy-extensions or gtk-extensions-legacy, if it was mate-only I would have chosen mate-gtk-extensions for the name.

**Comment by vkareh** [comment]
I've started work on fixing this. I'm basically re-implementing some of the lost functionality from the `GtkIconInfo`, which GTK dropped (they basically kept the functions, but they all return `NULL`).

![screenshot at 2019-03-02 12-04-07](https://user-images.githubusercontent.com/259780/53685059-a7162580-3ce3-11e9-9121-e358a6545984.png)

It's currently full of hacks that probably won't work across themes. GTK really dropped _a lot_ of the functionality, so I'm implementing the bare minimum to get it working

**Comment by Joz3d** [comment]
So sad what they've done.  I want to vocalize my support for your bringing back this lost functionality.  Thank you.

**Comment by N0rbert** [comment]
Still problem for Caja 1.24 as in upcoming Ubuntu MATE 20.10.

**Comment by N0rbert** [comment]
Do we have any progress on this bug?

**Comment by vkareh** [comment]
> Do we have any progress on this bug?

No, but I just rebased and pushed my old branch from 3 years ago that has some of the work done: https://github.com/mate-desktop/caja/tree/fix-file-previews

**Comment by N0rbert** [comment]
Thanks @vkareh , I compiled this branch locally then tried `echo text > ~/file.txt` with `caja -q; caja ~` .
![caja-patched](https://user-images.githubusercontent.com/309456/148993819-671c3f51-bd31-4cc0-8f42-5b91592957d8.png)
I see only two letters of "text" - "te".
